osbenchmark/metrics.py

```diff
@@ -52,20 +52,23 @@ def __init__(self, client, cluster_version=None):
         self._client = client
         self.logger = logging.getLogger(__name__)
         self._cluster_version = cluster_version
+        self._cluster_distribution = None
 
     # TODO #653: Remove version-specific support for metrics stores before 7.0.0.
     def probe_version(self):
         info = self.guarded(self._client.info)
         try:
             self._cluster_version = versions.components(info["version"]["number"])
+            self._cluster_distribution = info["version"]["distribution"]
```

Comment:
If I curl an ES cluster it doesn't return a distribution in the version
```
"version" : {
    "number" : "7.7.0",
```
but it looks like all OpenSearch clusters do (tested Amazon OpenSearch Service and self-managed)
```
"version" : {
    "distribution" : "opensearch",
    "number" : "1.1.0",
```

The code as is will throw a KeyError for Elasticsearch clusters, can we refactor to the following?
```
def probe_version(self):
    info = self.guarded(self._client.info)
    try:
        self._cluster_version = versions.components(info["version"]["number"])
    except BaseException:
        msg = "Could not determine version of metrics cluster"
        self.logger.exception(msg)
        raise exceptions.BenchmarkError(msg)

    try:
        self._cluster_distribution = info["version"]["distribution"]
    except BaseException:
        msg = "Could not determine distribution of metrics cluster, assuming elasticsearch"
        self.logger.exception(msg)
        self._cluster_distribution = "elasticsearch"
```

Reply:
I missed that. Just pushed the correction